Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericTestParameterConverter.cs

```diff
@@ -1,69 +1,108 @@
+    public override GenericTestPageParameterDto Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+      if (reader.TokenType != JsonTokenType.StartObject)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.StartObject, reader);
+
+      reader.Read();
+
+      if (reader.TokenType != JsonTokenType.PropertyName)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.PropertyName, reader);
+
+      if (reader.GetString() != "status")
+        throw CreateExceptionForExpectedTokenStringValue ("status", reader);
+
+      reader.Read();
+      var status = (GenericTestPageStatus) reader.GetInt32();
+
+      reader.Read();
+
+      if (reader.TokenType != JsonTokenType.PropertyName)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.PropertyName, reader);
+
+      if (reader.GetString() != "parameters")
+        throw CreateExceptionForExpectedTokenStringValue ("parameters", reader);
+
+      reader.Read();
+
+      var dictionary = JsonSerializer.Deserialize<Dictionary<string, IList<string>>> (ref reader);
+
+      reader.Read();
+
+      if (reader.TokenType != JsonTokenType.EndObject)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.EndObject, reader);
+
+      return new GenericTestPageParameterDto (
+          status,
+          dictionary?.ToDictionary (
+              kvp => kvp.Key,
+              kvp => new GenericTestPageParameter (kvp.Key, kvp.Value.ToArray()))
+          ?? new Dictionary<string, GenericTestPageParameter>());
+    }
+
+    public override void Write (Utf8JsonWriter writer, GenericTestPageParameterDto value, JsonSerializerOptions options)
     {
-      var status = (GenericTestPageStatus) (int) dictionary["status"];
-      var parameters = GetDeserializedParameterDictionary (dictionary["parameters"]);
+      writer.WriteStartObject();
+
+      writer.WriteNumber ("status", (int) value.Status);
+
+      writer.WriteStartObject ("parameters");
+      foreach (var parameter in value.Parameters)
+      {
+        writer.WritePropertyName (parameter.Key);
+        JsonSerializer.Serialize (writer, parameter.Value.Arguments);
+      }
+      writer.WriteEndObject();
 
-      return new GenericTestPageParameterDto (status, parameters);
+      writer.WriteEndObject();
     }
 
-    /// <inheritdoc />
-    public override IDictionary<string, object> Serialize (object obj, JavaScriptSerializer serializer)
+    public override bool CanConvert (Type typeToConvert)
     {
-      var information = (GenericTestPageParameterDto) obj;
-      return new Dictionary<string, object>
-             {
-                 { "status", (int) information.Status },
-                 { "parameters", information.Parameters.ToDictionary (pair => pair.Key, pair => pair.Value.Arguments) }
-             };
+      return typeToConvert == typeof (GenericTestPageParameterDto);
```

Comment:
This should be redundant:
_Override the CanConvert method only if necessary. The default implementation returns true when the type to convert is of type T. Therefore, converters that support only type T don't need to override this method (...)._ ([Source](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-5-0#steps-to-follow-the-basic-pattern))

Except, of course, we want it to be as explicit as possible.